utils/install.py

```diff
@@ -0,0 +1,1585 @@
+                return -1
+            else:
+                return (a > b) - (a < b)
+        else:
+            a = self._preprocess(self.version, separator, ignorecase)
+            b = self._preprocess(version2, separator, ignorecase)
+            try:
+                return (a > b) - (a < b)
+            except:
+                return False
+
+
+SUPPORTED_PLATFORM_MACHINE = {
+    "Linux": ["x86_64", "amd64", "arm64", "armv8", "aarch64"],
+    "Darwin": ["x86_64", "arm64", "arm"],
+}
+
+SUPPORTED_MIN_VERSION = {
+    "Linux" + "x86_64": VersionString("0.8.0"),
+    "Linux" + "amd64": VersionString("0.8.0"),
+    "Linux" + "arm64": VersionString("0.8.0"),
+    "Linux" + "armv8": VersionString("0.8.1"),
+    "Linux" + "aarch64": VersionString("0.8.1"),
+    "Darwin" + "x86_64": VersionString("0.8.2"),
+    "Darwin" + "arm64": VersionString("0.8.2"),
+    "Darwin" + "arm": VersionString("0.8.2"),
+}
+
+WASMEDGE = "WasmEdge"
+WASMEDGE_UNINSTALLER = "WasmEdge_Uninstaller"
+TENSORFLOW = "tensorflow"
+TENSORFLOW_LITE = "tensorflow_lite"
+TENSORFLOW_DEPS = "tensorflow_deps"
+TENSORFLOW_LITE_DEPS = "tensorflow_lite_deps"
+TENSORFLOW_TOOLS = "tensorflow_tools"
+IMAGE = "image"
+IMAGE_DEPS = "image_deps"
+EXTENSIONS = [TENSORFLOW, IMAGE]
+
+DEPENDENCIES = {
+    TENSORFLOW: [
+        TENSORFLOW_DEPS,
+        TENSORFLOW_TOOLS,
+    ],
+    IMAGE: [
+        IMAGE_DEPS,
+    ],
+}
+
+SUPPORTED_EXTENSIONS = {
+    "Linux" + "x86_64": EXTENSIONS,
+    "Linux" + "amd64": EXTENSIONS,
+    "Linux" + "arm64": EXTENSIONS,
+    "Linux" + "armv8": EXTENSIONS,
+    "Linux" + "aarch64": EXTENSIONS,
+    "Darwin" + "x86_64": EXTENSIONS,
+    "Darwin" + "arm64": [],
+    "Darwin" + "arm": [],
+}
+
+SUPPORTED_EXTENSIONS_VERSION = {
+    "Linux" + "x86_64" + TENSORFLOW: VersionString("0.8.1"),
+    "Linux" + "x86_64" + IMAGE: VersionString("0.8.1"),
+    "Linux" + "x86_64" + IMAGE_DEPS: VersionString("0.8.2"),
+    "Linux" + "amd64" + TENSORFLOW: VersionString("0.8.1"),
+    "Linux" + "amd64" + IMAGE: VersionString("0.8.1"),
+    "Linux" + "arm64" + TENSORFLOW: VersionString("0.8.1"),
+    "Linux" + "arm64" + IMAGE: VersionString("0.8.1"),
+    "Linux" + "armv8" + TENSORFLOW: VersionString("0.8.1"),
+    "Linux" + "armv8" + IMAGE: VersionString("0.8.1"),
+    "Linux" + "aarch64" + TENSORFLOW: VersionString("0.9.1-beta.1"),
+    "Linux" + "aarch64" + IMAGE: VersionString("0.9.1-beta.1"),
+    "Darwin" + "x86_64" + TENSORFLOW: VersionString("0.10.0-alpha.1"),
+    "Darwin" + "x86_64" + IMAGE: VersionString("0.10.0-alpha.1"),
+    "Darwin" + "arm64" + TENSORFLOW: VersionString("0.10.0-alpha.1"),
+    # "Darwin" + "arm64" + IMAGE: VersionString("0.8.1"),
+    "Darwin" + "arm" + TENSORFLOW: VersionString("0.10.0-alpha.1"),
+    # "Darwin" + "arm" + IMAGE: VersionString("0.8.1"),
+}
+
+WASI_NN_OPENVINO = "wasi_nn-openvino"
+
+PLUGINS = [WASI_NN_OPENVINO]
+
+SUPPORTED_PLUGINS = {
+    "Linux" + "x86_64" + WASI_NN_OPENVINO: VersionString("0.10.1-alpha.1"),
```

Comment:
Since we are using the metadata to save all versions of plugins, is it still needed to list these plugins in the python file?

Reply:
Yes, we don't.